cpp/src/gandiva/engine.cc

```diff
@@ -218,6 +203,45 @@ void* Engine::CompiledFunction(llvm::Function* irFunction) {
   return execution_engine_->getPointerToFunction(irFunction);
 }
 
+void Engine::AddGlobalMappingForFunc(const std::string& name, llvm::Type* ret_type,
+                                     const std::vector<llvm::Type*>& args,
+                                     void* function_ptr) {
+  auto prototype = llvm::FunctionType::get(ret_type, args, false /*isVarArg*/);
+  auto fn = llvm::Function::Create(prototype, llvm::GlobalValue::ExternalLinkage, name,
+                                   module());
+  execution_engine_->addGlobalMapping(fn, function_ptr);
+}
+
+void Engine::AddGlobalMappings() {
+  std::vector<llvm::Type*> args;
+
+  // gdv_fn_like_utf8_utf8
+  args = {types().i64_type(), types().i8_ptr_type(), types().i32_type(),
```

Comment:
I'm curious: is it possible to declare those signatures in the same file as the function they describe? It would probably makes things easier to follow and maintain.

Reply:
Also, perhaps some kind of registry mechanism would be more future-proof.